pkg/apis/openebs.io/v1alpha1/new_cstor_pool.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=cstorpool
+
+// NewTestCStorPool describes a cstor pool resource created as custom resource.
+type NewTestCStorPool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NewCStorPoolSpec `json:"spec"`
+	Status CStorPoolStatus  `json:"status"`
+}
+
+// CStorPoolSpec is the spec listing fields for a CStorPool resource.
+type NewCStorPoolSpec struct {
```

Comment:
Instead of creating new one, can't we use `PoolSpec` from cstorpoolcluster?

Reply:
It will conflict..

Reply:
Ok